### ISSUE TITLE
Send an email to commentor/author of a post for every update to the post

### ISSUE THREAD

**Issue by idjevm** [FightPandemics/FightPandemics#945]
As a commentor/author of a post,
I want to receive the notifications on every comments posted/edits made from each of the commentor,
so that I can close my requirements.

Acceptance Criteria:
- [ ] All the users receive email
- [ ] The email is triggered for posting, commenting, editing posts, editing comments
- [ ] When the author is org profile, the mail must reach on the org email as stored on org profile
- [ ] The mail must have standard template with the message "You have received a comment on your post"
- [ ] The mail must contain a link to the post

Note this is only for logged in users.

**Comment by idjevm** [comment]
Hey @joshmorel so I will be taking into this issue and investigate it in about 6hrs (after I wake up). I was wondering what are your thoughts when it comes to collecting the emails / commenter's id (maybe having an endpoint?) so that way we can set up some sort of communication between the commenter and the post author. So far I think @ppagarawal wants to have a feature such as GitHub where it notified the user of a new comment, and those users get notified as well.

**Comment by joshmorel** [comment]
@idjevm @ppagarawal  Hmm... we have to think about this a bit more. We can notify the poster that they received a comment but sharing the email - right now this doesn't happen with facebook/twitter etc - the email/phone number remains private.

I think in such a case we would need to add an option when someone comments like "share my email with author" or something.  

Prioritizing DMs for V1.1 might be a good idea but we'll also need some kind of notification system in general (both email & internal to the app) that can also be configured.

Also we need to use Amazon SES for now. Our current SendGrid account level is not good for transactional emails - with auth emails many went to spam and some were even blocked by the email host. The same will happen here.

**Comment by joshmorel** [comment]
And I think this is hard IMHO or at least somewhere between medium and hard.

**Comment by idjevm** [comment]
Yeah, I totally agree with you Josh, probably at a hard level. I gave it a thought and started draft ideas, but after speaking with @ppagarawal, decided to put it on hold on my end since it seems that this needs to be done ASAP and there might be other engineers who might develop it faster.  (this was before the call)

**Comment by joshmorel** [comment]
@idjevm @manualzuru @ppagarawal I'm a little unclear with this two points:
cc @mannykary 

> All the users receive email
> The email is triggered for posting, commenting, editing posts, editing comments

Why would the get an email for editing a post? I would suggest - since we don't have opt in/out settings - we limit to two scenarios to start with:

1. Post author emailed when comment made
2. Comment author emailed when comment replied to (we don't have replies yet but should soon - will add a F/E ticket for this if there isn't one already)

More than this may get spammy especially since we won't have notification settings.

**Comment by idjevm** [comment]
Yeah I wanted to keep it simple to just notify that a user that there is a new comment, but I believe @ppagarawal wanted to have that implemented as well? Not sure as if we keep overthinking this feature, it will keep getting complex and complex...

**Comment by ppagarawal** [comment]
@idjevm @joshmorel The feature is complete only when the request and supply is matched. By notifying only 1 end, we don't really close the loop. This was the reason why I had recommended to have email notifications for comments and posts to the author of the post and the commentor. 

I understand that the implementation will be difficult but I think that a partial implementation will not fulfill the requirements from a user. As, the user will expect emails for any changes which happen on his post/comment. 

So, lets take this step-wise. Let the author know via that there is a comment on his post everytime it is commented on. I am working to create a complete pod on this feature to have clearer requirements.

**Comment by idjevm** [comment]
Yeah that makes sense, I guess? I mean until we don't have notifications in place, then I guess the best way will be just to email notifications every time there is a change. I don't think we should do this for `on comment edit` or `on post edit` notification though, I don't think that's necessary. Think of it like a slack thread, do you get notified if I change edit the original message? I think it is a no, and I think we will be going over the top if we try to address that to notify `on edit`

**Comment by ppagarawal** [comment]
Let's begin on this to send notification to the author for every change in state for the first release. It would be great to get this moving rather than spending time only on discussions.

**Comment by idjevm** [comment]
I mean the discussion is needed, else we will not consider things that otherwise would've been ignored. For example, had I started just implementing this without discussing, I would've started to find these blockers and other things that I did not initially consider that are concerning.  Let's hope that for now, we have a better idea of how this will move forward.

**Comment by doheekim93** [comment]
@skrithiga @nfranklin9 I assigned this ticket to you guys, so you can keep track of all the open github tickets relating to notifications! You can unassign yourselves and assign engineers when you find someone to work on this ticket 😄

**Comment by emtay** [comment]
@skrithiga @nfranklin9 it may be easier to close this ticket and open a new one with the same tags and your most recent requirements for email notifications?

from thread above, i agree with prateek: 

- author of post gets emails for all comments to the post from other users (but no email needed for when they add their post or edit their post)
- commenter on post get notification of future comments on same post (whether from author or other user)

(no email for editing comments)
